pkg/scheduler/scheduler.go

```diff
@@ -454,6 +454,21 @@ func (sched *Scheduler) scheduleOne() {
 	// Synchronously attempt to find a fit for the pod.
 	start := time.Now()
 	pluginContext := framework.NewPluginContext()
+
+	// Run "prefilter" plugins.
+	prefilterStatus := fwk.RunPrefilterPlugins(pluginContext, pod)
```

Comment:
I don't think we should call Prefilter here. Prefilter, Filter, PostFilter, Scoring, and Normalize Scoring should be invoked by genericScheduler. We should pass pluginContext to genericScheduler and let it run these plugins. We should also start thinking about providing some of the object listers via FrameworkHandle so that plugins can use them.

Reply:
Can you elaborate why this should be called from generic scheduler?

> We should also start thinking about providing some of the object listers 
> via FrameworkHandle so that plugins can use them.
Should this be in a separate PR?

Reply:
ok, taking a closer look at the code I think it make sense to move prefilter inside generic scheduler since all the other extensions points you mention can only exist there, and it make sense to put prefilter in the same place as filter and post filter.

Reply:
> Should this be in a separate PR?

Yes, absolutely. In fact, it does not make sense to put those changes in this PR. The other PR should probably be postponed until we start converting predicate/priority functions.

> ok, taking a closer look at the code I think it make sense to move prefilter inside generic scheduler since all the other extensions points you mention can only exist there, and it make sense to put prefilter in the same place as filter and post filter.

That's right. That's exactly the reason.

Reply:
One more thing that I would like to point out for the record (and does not apply to this PR, but is related to our discussion above) is that we should think about removing `genericScheduler` and replacing it with the Framework.

Reply:
Done moving to genericScheduler. PTAL.

I agree with removing genericScheduler.

Reply:
May I ask why we should removing genericScheduler?

Reply:
The generic scheduler functionality is being moved to the scheduler framework.